src/QirRuntime/lib/QIR/bridge-qis.ll

```diff
@@ -348,6 +356,80 @@ define double @__quantum__qis__arctan2__body(double %y, double %x) {  ; Q#: func
   ret double %result
 }
 
+; function Sin (theta : Double) : Double
+; https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.math.sin
+define double @__quantum__qis__sin__body(double %theta) {       ; https://en.cppreference.com/w/cpp/numeric/math/sin
+    %result = call double @llvm.sin.f64(double %theta)          ; https://llvm.org/docs/LangRef.html#llvm-sin-intrinsic
+    ret double %result
+}
+
+; function Cos (theta : Double) : Double
+; https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.math.cos
+define double @__quantum__qis__cos__body(double %theta) {       ; https://en.cppreference.com/w/cpp/numeric/math/cos
+    %result = call double @llvm.cos.f64(double %theta)          ; https://llvm.org/docs/LangRef.html#llvm-cos-intrinsic
+    ret double %result
+}
+
+; function Tan (theta : Double) : Double
+; https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.math.tan
+define double @__quantum__qis__tan__body(double %theta) {       ; https://en.cppreference.com/w/cpp/numeric/math/tan

```

Comment:
I don't know how std::tan is implemented and how smart it is, but it's possible that the library implementation might have higher accuracy for angles that are close to 0 and Pi/2 (rather than doing division they might go with other ways to compute the result).

Reply:
#532.
Thanks for the insight.